docs/topics/certificaterotation.md

```diff
@@ -0,0 +1,52 @@
+# Cluster certificate rotation
+
+Instructions on rotating TLS CA and certificates for an AKS Engine cluster.
+
+## Prerequesites
+
+- Kubernetes version 1.8.0 or later is required
+- The etcd members MUST be in a healthy state before rotating the CA and certs.
+- The `_output` directory for your cluster deployment, including a working SSH key for accessing the master nodes and `kubeconfig`.
+
+## Preparation
+
+**CAUTION**: Before rotation, it's preferrable to back up your cluster:
+
+1. On an etcd node, take a backup of the current state.
+2. //TODO recommend etcd backup tool, extract the existing state from etcd. Copy this back to your working machine as a precaution, in case the control plane goes down.
+3. Keep a copy of your current _output directory.
+
+## Rotation
+
+**CAUTION**: Rotating certificates will cause cluster downtime.
+
+run `aks-engine rotate-certs`
+
+`aks-engine rotate-certs` will:
+
+- Use the exisiting certificate generation code to generate new certificates, using the same certificate properties and default expiration as new certificates generated for new clusters.
+- Rotate etcd CA and certificates in all of the master nodes.
+- From the first master node, rotate apiserver certificates in the nodes.
+- Update the kubeconfig and rotate the kubelet certificates.
+
+## Reboot Cluster
+
+After updating the CA and certs of the API server, we need to restart all the pods to ensure they refresh their service account.
+
+This can be done by rebooting all the nodes in the cluster.
+
+// TODO: add `aks-engine` reboot command
+
+## Verification
+
+After the above steps, you can verify the success of the CA and certs rotation:
+
+- Old  `kubeconfig`  should  **NOT**  be able to contact the API server, however the new kubeconfig should be able to talk to it.
```

Comment:
We might consider writing a general purpose script to do this validation (might be useful apart from cert rotation context):

`aks-engine validate --kubeconfig=<path-to-kube-config> --api-model=<path-to-api-model>`

Or maybe we can parse the template/params to get the validation input (number of nodes, etc), either via a local file or via running an `az`-like command against the resource group.

Reply:
Good idea, that would be phase 2 though. Phase 1 will focus on the cert rotation itself.

Reply:
@jackfrancis what would that validate command do exactly we can spec it for phase two or independently if it makes more sense.